packages/spectral/src/generators/componentManifest/getInputs.ts

```diff
@@ -90,7 +92,29 @@ export const INPUT_TYPE_MAP: Record<InputFieldDefinition["type"], InputType> = {
 };
 
 const getInputType = (input: ServerTypeInput) => {
-  return input.model
+  const valueType = input.model
     ? input.model.map((choice) => `"${choice.value}"`).join(" | ")
     : INPUT_TYPE_MAP[input.type as InputFieldDefinition["type"]] || "never";
+
+  if (isKeyValueList(input)) {
+    return `Record<string, ${valueType}>`;
+  }
+
+  if (isValueList(input)) {
+    return `${valueType}[]`;
+  }
+
+  return valueType;
+};
+
+const isKeyValueList = (
+  input: ServerTypeInput
+): input is ServerTypeInput & { collection: "keyvaluelist" } => {
+  return input.collection === "keyvaluelist";
+};
+
+const isValueList = (
+  input: ServerTypeInput
+): input is ServerTypeInput & { collection: "valuelist" } => {
+  return input.collection === "valuelist";
```

Comment:
This is fine to do but we don't really need a type guard here as we're just checking the value of the optional `collection` property. Could change the above `if` statements to the following and remove these functions:

```ts
if (input.collection === "keyvaluelist") {
  return `Record<string, ${valueType}>`;
}
```